src/System.IO.Abstractions.TestingHelpers/MockFile.cs

```diff
@@ -203,8 +203,24 @@ public override bool Exists(string path)
                 return false;
             }
 
-            var file = mockFileDataAccessor.GetFile(path);
-            return file != null && !file.IsDirectory;
+            if (path.Trim() == string.Empty)
+            {
+                return false;
+            }
+
+            try
+            {
+                mockFileDataAccessor.PathVerifier.IsLegalAbsoluteOrRelative(path, nameof(path));
+
+                var file = mockFileDataAccessor.GetFile(path);
+                return file != null && !file.IsDirectory;
+            }
+            catch (ArgumentException) { }
```

Comment:
@masood-haloratech Please add clarifying comments in the empty catch blocks so that it's clear why we don't handle the exceptions.